sdk/build-scripts/src/build-package.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { promises as fs, existsSync } from 'fs';
+import * as path from 'path';
+import { build } from 'esbuild';
+import { execSync } from 'child_process';
+
+interface PackageJSON {
+	name: string;
+	exports?: Record<string, string | Record<string, string>>;
+}
+
+const ignorePatterns = [/\.test.ts$/];
+
+buildPackage().catch((error) => {
+	console.error(error);
+	process.exit(1);
+});
+
+async function buildPackage() {
+	const allFiles = await findAllFiles(path.join(process.cwd(), 'src'));
+	const packageJson = await readPackageJson();
+	await clean();
+	await buildCJS(allFiles);
+	await buildESM(allFiles);
+	await buildImportDirectories(packageJson);
+}
+
+async function findAllFiles(dir: string, files: string[] = []) {
+	const dirFiles = await fs.readdir(dir);
+	for (const file of dirFiles) {
+		const filePath = path.join(dir, file);
+		const fileStat = await fs.stat(filePath);
+		if (fileStat.isDirectory()) {
+			await findAllFiles(filePath, files);
+		} else if (!ignorePatterns.some((pattern) => pattern.test(filePath))) {
+			files.push(filePath);
+		}
+	}
+	return files;
+}
+
+async function clean() {
+	await createEmptyDir(path.join(process.cwd(), 'dist'));
+}
+
+async function buildCJS(entryPoints: string[]) {
+	await build({
+		format: 'cjs',
+		logLevel: 'error',
+		target: 'es2020',
+		entryPoints,
```

Comment:
This basically is every single file right? Is there a reason why you're using esbuild vs just farming this out to `tsc`? I don't think this is bad more (tsc is not a great transpiler) just trying to understand your thinking here.

Reply:
mostly performance/future flexibility.  esbuild is a better transpiler.  tsc would work fine, but even typescript itself build its releases with esbuild

Reply:
Yeah sounds good to me